### PR TITLE
Added type flags to nodes

### PR DESCRIPTION
**Pull request by Daniel-Ayala**
get_js_struct now add some additional flags to the json:
 * 'type' determines whether the node is a left node, a right one, or a root node.
 * 'is_prediction' determines whether the node is a dredicted class or not. That is, whether or not it is a leaf node.

These flags can be used to further stylize the d3 tree, so that decision nodes have a different look, etc.

### CODE REVIEW

File: eurekatrees.py [hunk 1]
```diff
@@ -81,11 +81,15 @@ def get_js_struct(self, node, node_dict=None):
             node_dict = {'name': node.data, 'children': []}
         if node is not None:
             if node.left:
-                new_node_dict_left = {'name': node.left.data, 'children': []}
+                new_node_dict_left = {'name': node.left.data, 'type':'left', 'is_prediction':False 'children': []}
```
Comment (ChuckWoodraska) [inline eurekatrees.py]: Needs comma after 'is_prediction':False. This is an error that needs to be fixed before I can accept your pull request.

File: eurekatrees.py [hunk 2]
```diff
@@ -81,11 +81,15 @@ def get_js_struct(self, node, node_dict=None):
...
                 node_dict['children'].append(self.get_js_struct(node.left, new_node_dict_left))
             if node.right:
-                new_node_dict_right = {'name': node.right.data, 'children': []}
+                new_node_dict_right = {'name': node.right.data, 'type':'right', 'is_prediction':False 'children': []}
```
Comment (ChuckWoodraska) [inline eurekatrees.py]: Needs comma after 'is_prediction':False. This is an error that needs to be fixed before I can accept your pull request.

**Review by ChuckWoodraska — CHANGES_REQUESTED** [review]
Please fix the issues noted and repush to this branch.

**Comment by Daniel-Ayala** [comment]
fixed

**Review by ChuckWoodraska — APPROVED** [review]
Looks good.

**Pull request merged**